src/index.js

```diff
@@ -352,135 +323,106 @@ var patch = function(parent, node, oldVNode, newVNode, listener, isSvg) {
   return (newVNode.node = node)
 }
 
-var propsChanged = function(a, b) {
+var propsChanged = (a, b) => {
   for (var k in a) if (a[k] !== b[k]) return true
   for (var k in b) if (a[k] !== b[k]) return true
 }
 
-var getTextVNode = function(node) {
-  return typeof node === "object" ? node : createTextVNode(node)
-}
-
-var getVNode = function(newVNode, oldVNode) {
-  return newVNode.type === LAZY_NODE
-    ? ((!oldVNode || !oldVNode.lazy || propsChanged(oldVNode.lazy, newVNode.lazy))
-        && ((oldVNode = getTextVNode(newVNode.lazy.view(newVNode.lazy))).lazy =
-          newVNode.lazy),
-      oldVNode)
-    : newVNode
-}
-
-var createVNode = function(name, props, children, node, key, type) {
-  return {
-    name: name,
-    props: props,
-    children: children,
-    node: node,
-    type: type,
-    key: key
-  }
-}
-
-var createTextVNode = function(value, node) {
-  return createVNode(value, EMPTY_OBJ, EMPTY_ARR, node, undefined, TEXT_NODE)
-}
-
-var recycleNode = function(node) {
-  return node.nodeType === TEXT_NODE
-    ? createTextVNode(node.nodeValue, node)
+var maybeVNode = (newVNode, oldVNode) =>
+  newVNode !== true && newVNode !== false && newVNode
+    ? typeof newVNode.tag === "function"
+      ? ((!oldVNode ||
+          oldVNode.memo == null ||
+          propsChanged(oldVNode.memo, newVNode.memo)) &&
+          ((oldVNode = newVNode.tag(newVNode.memo)).memo = newVNode.memo),
+        oldVNode)
+      : newVNode
+    : text("")
+
+var createVNode = (type, props, children, node, key, tag) => ({
+  type: type,
+  props: props,
+  children: children,
+  node: node,
+  tag: tag,
+  key: key,
+})
+
+var recycleNode = (node) =>
+  node.nodeType === TEXT_NODE
+    ? text(node.nodeValue, node)
     : createVNode(
         node.nodeName.toLowerCase(),
         EMPTY_OBJ,
         map.call(node.childNodes, recycleNode),
         node,
-        undefined,
-        RECYCLED_NODE
+        null,
+        RECYCLED
       )
-}
 
-export var Lazy = function(props) {
-  return {
-    lazy: props,
-    type: LAZY_NODE
-  }
-}
+var memo = (view, props) => ({ memo: props, tag: view })
 
-export var h = function(name, props) {
-  for (var vdom, rest = [], children = [], i = arguments.length; i-- > 2; ) {
-    rest.push(arguments[i])
-  }
+var text = (value, node) =>
+  createVNode(value, EMPTY_OBJ, EMPTY_ARR, node, null, TEXT_NODE)
 
-  while (rest.length > 0) {
-    if (isArray((vdom = rest.pop()))) {
-      for (var i = vdom.length; i-- > 0; ) {
-        rest.push(vdom[i])
-      }
-    } else if (vdom === false || vdom === true || vdom == null) {
-    } else {
-      children.push(getTextVNode(vdom))
-    }
-  }
+var h = (type, props, children) =>
+  createVNode(
+    type,
+    props,
+    isArray(children) ? children : children == null ? EMPTY_ARR : [children],
+    null,
+    props.key
```

Comment:
In Babel's JSX, the `props` may be null, so a null error occurs.

Reply:
Thanks for pointing that out. We're dropping out-of-the-box JSX support. So, you'll need a dedicated babel plugin to compile JSX to Hyperapp's hyperscript or use a tagged template literal function like @zaceno's https://github.com/zaceno/hyperlit.

Reply:
@jorgebucaran @diontools 

Technically, you could still continue to use `babel-plugin-transform-react-jsx` for your jsx. You just can't use hyperapp´s built in `h` pragma any more. You need a custom pragma as a go-between what JSX compiles to, and what the new `h` will accept

Reply:
Ah, yes, that'd work too!

Reply:
@jorgebucaran @zaceno Thanks!